zxlive/edit_panel.py

```diff
@@ -200,20 +215,29 @@ def paste_graph(self, graph: GraphT) -> None:
     def delete_selection(self) -> None:
         selection = list(self.graph_scene.selected_vertices)
         selected_edges = list(self.graph_scene.selected_edges)
-        if not selection and not selected_edges: return
+        rem_vertices = selection.copy()
+        for v in selection:
+            if vertex_is_w(self.graph_scene.g.type(v)):
+                for n in self.graph_scene.g.neighbors(v):
+                    if self.graph_scene.g.edge_type((v, n)) == EdgeType.W_IO:
+                        rem_vertices.append(n)
+        for v1, v2 in selected_edges:
+            if self.graph_scene.g.edge_type((v1, v2)) == EdgeType.W_IO:
+                rem_vertices.extend([v1, v2])
```

Comment:
Maybe we should make it that `W_IO` edges cannot be selected? This could be done easily by calling
```python
self.setFlag(QGraphicsItem.GraphicsItemFlag.ItemIsSelectable, self.g.edge_type(self.e) != EdgeType.W_IO)
```
in `EItem.refresh()`

Reply:
I think this is a great idea but why should it be in `EItem.refresh()` instead of `__init__()`?

Reply:
I have added it in `__init__` and it seems to work

Reply:
When the edge type changes for some reason, selection should be reanabled. Not sure if IO edges can change (maybe a rewrite or though the user), but we could put it into `refresh` just to be safe

Reply:
They should never change but happy to move it to `refresh` to be safe.